File: First and Last.py
# cook your dish here

t = int(input())

for _ in range(t):
    
    n = int(input())
    a = list(map(int , input().split()))
    
    maxim = a[0] + a[-1]
    
    for i in range(n): # n Right Rotations
        
        # Right Rotating
        x = a.pop(-1)
        a.insert(0, x)
        
        elem = a[0] + a[-1]
        if elem > maxim:
            maxim = elem
    
    print(maxim)

File: First and Last.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    
    int t;
    int n;
    
    int i;
    
    cin >> t;
    
    for(int ii = 1; ii <= t; ii++)
    {
    
        cin >> n;
        
        int arr[n];
        
        for(i=0; i<n; i++)
            cin >> arr[i];
        
        int max = arr[0] + arr[n-1];
        
        for(int _ = 1; _<= n; _++) // n Right Rotations
        {
            // Right Rotating
            int x = arr[n-1];
            for(i = n-1; i>= 1; i--)
            {
                arr[i] = arr[i-1];
            }
            arr[0] = x;
            
            int elem = arr[0] + arr[n-1];
            if (elem > max)
                max = elem;
            
        }
        
        cout << max << endl;
    
    }
    
	return 0;
}
